urldeb33fd2fa_friedemann_distripedia_tgz-pJ8-org_json_TestJ8/src/grinder/client/ClientGUI.java

```diff
@@ -0,0 +1,446 @@
+				btn_connect.setAction(actionDisconnect);
+			}
+			btn_connect.setEnabled(true);
+		}
+	};
+
+	/** The action disconnect. */
+	public Action actionDisconnect = new AbstractAction("disconnect") {
+		@Override
+		public void actionPerformed(final ActionEvent e) {
+
+			if (c.isConnected())
+			{
+				final int dlg =
+						confirmDialog("really disconnect?",
+								"Are you sure you want to disconnect this client? "
+										+ "This results in a complete loss of data this "
+										+ "client has gathered so far.");
+				if (dlg == JOptionPane.OK_OPTION)
+				{
+					c.disconnectFromController();
+				}
+			}
+			field_ip.setEnabled(true);
+			btn_connect.setAction(actionConnect);
+
+		}
+	};
+
+	/**
+	 * Static method - can be called from outer space to initiate generation of
+	 * frame. Constructor of this class is invoked within.
+	 *
+	 * @param title - name of the program
+	 */
+	public static void loadGUI(final String title) {
+		// try to set the UI to native win/mac layout
+		ClientGUI.setNativeLookAndFeel();
+
+		// setup a new frame
+		ClientGUI.mainframe =
+				new JFrame(title + ClientGUI.PROGRAM_VERSION + ClientGUI.TITLE_SUFFIX);
+		ClientGUI.mainframe.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+		ClientGUI.mainframe.setSize(ClientGUI.DEFAULT_WIDTH, ClientGUI.DEFAULT_HEIGHT);
+		ClientGUI.mainframe.setMinimumSize(new Dimension(ClientGUI.MINIMUM_WIDTH,
+				ClientGUI.MINIMUM_HEIGHT));
+
+		// instantiate a new gui panel here...
+		final JComponent guiContentPane = new ClientGUI();
+		guiContentPane.setOpaque(true);
+
+		// ...and connect it to the main frame
+		ClientGUI.mainframe.setContentPane(guiContentPane);
+
+		// resolve the screensize
+		final Toolkit toolkit = Toolkit.getDefaultToolkit();
+		final Dimension screenSize = toolkit.getScreenSize();
+
+		// center the frame
+		ClientGUI.mainframe.setLocation(
+				(screenSize.width - ClientGUI.mainframe.getWidth()) / 2,
+				(screenSize.height - ClientGUI.mainframe.getHeight()) / 2);
+		ClientGUI.mainframe.setVisible(true);
+	}
+
+	/**
+	 * Sets the status on the GUI
+	 *
+	 * @param state the new status
+	 */
+	public void setStatus(final ClientState state) {
+		lbl_status.setText(state.toString());
+	}
+
+	/**
+	 * Updates the gui's user count.
+	 */
+	public void updateUserCount() {
+
+		Integer uCount = 0;
+		uCount += c.userHTTP.size();
+		uCount += c.userWS.size();
+		uCount += c.userSTRATUS.size();
+
+		lbl_users.setText(uCount.toString());
+	}
+
+	/**
+	 * Increase request counter.
+	 */
+	public void increaseRequestCounter() {
+		Integer i = new Integer(lbl_requests.getText());
+		i++;
+		lbl_requests.setText(i.toString());
+	}
+
+	/**
+	 * Resets request counter.
+	 */
+	public void resetRequestCounter() {
+		lbl_requests.setText("0");
+	}
+
+	/**
+	 * Log message to the gui.
+	 *
+	 * @param msg the msg
+	 */
+	public synchronized void logMessage(final String msg) {
+		final Format formatter = new SimpleDateFormat(GrinderSettings.SIMPLE_DATE_FORMAT);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The SimpleDateFormat object was created without setting the time zone. Make sure that you want to use the default time zone or use letter z, Z or X in the pattern. Otherwise, if you do not call setTimeZone() on the created SimpleDateFormat object, you might get an unexpected date and time when using the object.